src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.GCDesc.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    partial class EETypeNode
+    {
+        private int GCDescSize
+        {
```

Comment:
It may also make sharing of the GCDesc encoder between EEType node, GCStatics node and eventual runtime type system easier. Basically, there would be two methods for encoding GCDescs - one for standard and one for arrays - both would take GCPointerMap and some fudge factors as input, and produce byte[] as output.